src/ape_run/_cli.py

```diff
@@ -76,13 +76,15 @@ def invoke(self, ctx: Context) -> Any:
             if ctx.params["interactive"]:
                 # Print the exception trace and then launch the console
                 # Attempt to use source-traceback style printing.
-                if not isinstance(err, ApeException) or not handle_ape_exception(
-                    err, [ctx.obj.project_manager.path]
-                ):
-                    err_info = traceback.format_exc()
-                    click.echo(err_info)
-
-                self._launch_console()
+                network_value = ctx.params.get("network") or networks.default_ecosystem.name
+                with networks.parse_network_choice(network_value):
```

Comment:
so the problem here with disconnecting and reconnecting is if the network context gets dumped

I think there has to be a way of extending the context such that the interactive mode stays under the main test context

Reply:
yep, ended up putting the provider in a classvar to 'recycle' it in the right conditions